File: src/plumbing_pub_sub/src/demo3_pub_person.cpp
#include "ros/ros.h"
#include "plumbing_pub_sub/Person.h"
/*
    发布方：发布人的数据
        1.包含头文件
        #include "plumbing_pub_sub/Person.h"
        2.初始化ros节点
        3.创建节点句柄
        4.创建发布者对象
        5.编写发布者逻辑，发布数据
*/

int main(int argc, char *argv[])
{
    setlocale(LC_ALL,"");
        //  2.初始化ros节点
    ros::init(argc,argv,"DaXia");   
        // 3.创建节点句柄
    ROS_INFO("发布方实现");
    ros::NodeHandle nh;
        // 4.创建发布者对象
    ros::Publisher pub =nh.advertise<plumbing_pub_sub::Person>("liaotian",10);
        // 5.编写发布者逻辑，发布数据
    // 创建发布数据
    plumbing_pub_sub::Person person;
    person.name = "张三";
    person.age  =10;
    person.height =100.0;
    // 设置发布频率
    ros::Rate rate(1);
    // 循环发布数据
    ros::Duration(3).sleep();
    while (ros::ok())
    {
        // 修改被发布的数据
        person.age +=1;
        pub.publish(person);
        ROS_INFO("发布的消息：%s,%d,%.2f",person.name.c_str(),person.age,person.height);
        rate.sleep();
        ros::spinOnce();
    }
    
    return 0;
}

File: src/plumbing_pub_sub/scripts/demo3_pub_person.py
#!/usr/bin/env python

import rospy
from plumbing_pub_sub.msg import Person
"""
        发布方实现
            1.导包
            2.初始化ros节点
            3.创建发布者实现
            4.发布消息
"""
if __name__=="__main__":

    rospy.init_node("DaMa")
    # 2.始化ros节点
    pub = rospy.Publisher("jiaoSheTou",Person,queue_size=10)
    # 3.创建发布者实现
    # 创建Person数据
    msg = Person()
    msg.name="奥特曼"
    msg.age =10
    msg.height=172.1
    # 创建rate对象
    rate = rospy.Rate(1)
    rospy.sleep(2)
    # 4.发布消息
    while not rospy.is_shutdown():
        msg.age +=1
        rospy.loginfo("发布数据：%s,%d,%.2f",msg.name,msg.age,msg.height)
        pub.publish(msg)
        rate.sleep()
    pass